File: practices/elixir/programming_elixir/mm/module_and_function-6.exs
# プログラミングElixir 6.5 ModuleAndFunctions-5

defmodule Chop do
  def guess(actual, a..b) when actual == div(a + b, 2) do
    IO.puts actual
  end

  def guess(actual, a..b) when actual > div(a + b, 2) do
    tmp = div(a + b, 2)

    IO.puts "Is it #{tmp}"

    guess(actual, tmp..b)
  end

  def guess(actual, a..b) when actual < div(a + b, 2) do
    tmp = div(a + b, 2)

    IO.puts "Is it #{tmp}"

    guess(actual, a..tmp)
  end
end

Chop.guess(273, (1..1000))

File: practices/elixir/programming_elixir/mm/module_and_function-6.rb
# プログラミングElixir 6.5 ModuleAndFunctions-5

class Chop
  def guess(actual, range)
    tmp = ((range.min + range.max) / 2).ceil

    puts "Is it #{tmp}"

    return puts actual if tmp.eql? actual

    if actual > tmp
      guess(actual, tmp..range.last)
    elsif actual < tmp
      guess(actual, range.first..tmp)
    end
  end
end

Chop.new.guess(273, (1..1000))
